File: frontend/types/dataset.ts
export interface Dataset {
  id: string;
  name: string;
  description: string | null;
  file_name: string;
  file_size: number;
  file_type: string;
  uploaded_by_email: string;
  created_at: string;
}

export interface DatasetListResponse {
  datasets: Dataset[];
}

export interface DownloadUrlResponse {
  url: string;
  expires_in: number;
}

export interface UploadResponse {
  dataset: Dataset;
  message: string;
}

File: backend/app/schemas/dataset.py
"""
Pydantic schemas for dataset upload/browse/download requests and responses.
"""

from typing import Optional

from pydantic import BaseModel


class DatasetResponse(BaseModel):
    id: str
    name: str
    description: Optional[str] = None
    file_name: str
    file_size: int
    file_type: str
    uploaded_by_email: str
    created_at: str


class DatasetListResponse(BaseModel):
    datasets: list[DatasetResponse]


class DownloadUrlResponse(BaseModel):
    url: str
    expires_in: int


class UploadResponse(BaseModel):
    dataset: DatasetResponse
    message: str
